core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -281,6 +282,44 @@ class EntityService(protected val userInfo: UserInfo, val dataSource: SlickDataS
   def batchUpsertEntities(workspaceName: WorkspaceName, entityUpdates: Seq[EntityUpdateDefinition], dataReference: Option[DataReferenceName], billingProject: Option[GoogleProjectId]): Future[Traversable[Entity]] =
     batchUpdateEntitiesInternal(workspaceName, entityUpdates, upsert = true, dataReference, billingProject)
 
+  def renameAttribute(workspaceName: WorkspaceName,
+                      entityType: String,
+                      oldAttributeName: AttributeName,
+                      attributeRenameRequest: AttributeRename): Future[Int] = {
+    getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write, Some(WorkspaceAttributeSpecs(all = false))) flatMap { workspaceContext =>
+
+      def validateNewAttributeName(dataAccess: DataAccess,
+                                   workspaceContext: Workspace,
+                                   entityType: String,
+                                   attributeName: AttributeName): ReadAction[Boolean] = {
+        dataAccess.entityQuery.doesAttributeNameAlreadyExist(workspaceContext, entityType, attributeName) map {
+          case Some(false) => false
+          case Some(true) => throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict,
+            s"${attributeName} already exists as an attribute name"))
```

Comment:
nit: use `${AttributeName.toDelimitedName(attributeName)}` in the error message instead of `${attributeName}` … the `toDelimitedName` will give a more user-friendly representation